.ci_support/build_all.py

```diff
@@ -0,0 +1,60 @@
+import conda_build.conda_interface
+import networkx as nx
+import conda_build.api
+from compute_build_graph import construct_graph
+import argparse
+import os
+
+def build_all(recipes_dir, arch):
+    folders = os.listdir(recipes_dir)
+    if not folders:
+        print("Found no recipes to build")
+        return
+    channel_urls=['local', 'conda-forge', 'defaults']
+    index = conda_build.conda_interface.get_index(channel_urls=channel_urls)
+    conda_resolve = conda_build.conda_interface.Resolve(index)
+
+    exclusive_config_file = os.path.join(conda_build.conda_interface.root_dir, 'conda_build_config.yaml')
+    platform =  get_host_platform()
+    variant_config_files = []
+    if platform == 'win':
+        script_dir = os.path.dirname(os.path.realpath(__file__))
+        variant_config_files = [os.path.join(script_dir, 'win{}.yaml'.format(arch))]
+
+    config = conda_build.api.Config(variant_config_files=variant_config_files, arch=arch,
+                                    exclusive_config_file=exclusive_config_file,
+                                    channel_urls=channel_urls)
+
+    worker={'platform': platform, 'arch': arch, 'label': '{}-{}'.format(platform, arch)}
+
+    G = construct_graph(recipes_dir, worker=worker, run='build', conda_resolve=conda_resolve,
+                        folders=folders, config=config)
+
+    order = list(nx.topological_sort(G))
+    order.reverse()
+
+    print('Computed that there are {} distributions to build from {} recipes'.format(len(order), len(folders)))
+    print("Resolved dependencies, will be built in the following order:")
+    print('    '+'\n    '.join(order))
+
+    for node in order:
+        conda_build.api.build(G.node[node]['meta'])
```

Comment:
@msarahan, for building recipes with multiple outputs, what should I do? Get the recipe paths from the meta data above, put it into a OrderedSet and pass the path to `conda_build.api.build`

Reply:
that's basically correct.  It's the collapsing that happens in https://github.com/conda/conda-concourse-ci/blob/master/conda_concourse_ci/compute_build_graph.py#L274

It's not perfect, mind you.  There are still some dependency relationships that are not properly handled.  See https://github.com/conda/conda-concourse-ci/issues/69

It would be worth giving your proposal a try, to see how well it works.  I suspect that there might be some crummy corner cases that mess things up, but your idea should work most of the time.